core/src/main/java/io/questdb/griffin/engine/table/AsyncGroupByRecordCursorFactory.java

```diff
@@ -395,7 +395,7 @@ private static void filterAndAggregate(
         final Function filter = atom.getFilter(slotId);
         final RecordSink mapSink = atom.getMapSink(slotId);
         try {
-            if (compiledFilter == null || frameSequence.getPageFrameAddressCache().hasColumnTops(task.getFrameIndex())) {
+            if (compiledFilter == null || frameMemory.hasColumnTops()) {
```

Comment:
Would be great to add a parallel test to make sure that we correctly decode and filter parquet row groups with coltops. Currently this code is untested for parquet.

Reply:
test coverage improved